server/src/test/java/org/apache/druid/metadata/IndexerSQLMetadataStorageCoordinatorTest.java

```diff
@@ -935,7 +935,12 @@ public void testTransactionalAnnounceFailDbNullWantNotNull() throws IOException
         new ObjectMetadata(ImmutableMap.of("foo", "bar")),
         new ObjectMetadata(ImmutableMap.of("foo", "baz"))
     );
-    Assert.assertEquals(SegmentPublishResult.fail("java.lang.RuntimeException: Failed to update the metadata Store. The new start metadata is ahead of last commited end state."), result1);
+    Assert.assertEquals(
+        SegmentPublishResult.fail(
+            "java.lang.RuntimeException: Failed to update the metadata Store."
```

Comment:
Better just throw a `DruidException` from method `commitSegmentsAndMetadata()`.

Reply:
Ack, I will update the callers to throw DruidException and update the tests accordingly.